lib/db_connection/ownership/proxy.ex

```diff
@@ -36,7 +36,7 @@ defmodule DBConnection.Ownership.Proxy do
 
     state = %{client: nil, timer: nil, holder: nil,
               timeout: timeout, interval: interval, poll: nil,
-              owner_ref: owner_ref, pool: pool, pool_ref: nil,
+              owner: {caller, owner_ref}, pool: pool, pool_ref: nil,
```

Comment:
This could be another entry in the map because this data never changes. `client` has a tuple to ensure "atomicity" so that we always set the whole client status and should be less likely to make a mistake. For owner/owner ref we never check both at same time so code would be clearer without the tuple.

Reply:
Yeah, but I also thought that if we were ever to change it, keeping them as a tuple guarantee we won't make stuff inconsistent. 

But I am indifferent. If you have a strong preference for one or another please let me know and I will change it.